packages/@aws-cdk/aws-ec2/lib/machine-image.ts

```diff
@@ -220,19 +220,38 @@ export enum OperatingSystemType {
   Windows,
 }
 
+/**
+ * Options for rendering UserData
+ */
+export interface UserDataOptions {
+  /**
+   * Whether to log commands as they're being executed
+   *
+   * @default true
+   */
+  verbose?: boolean;
+
+  /**
+   * Whether to stop executing as soon as an error is encountered
+   *
+   * @default false
```

Comment:
Wouldn't it make sense that this would be `true` by default?
Also, we should document what this means on each platform?

Reply:
Yes, we probably do want that, but then we have to do signaling very differently (which I think is still a good thing!)

Signaling is supposed to be used this way:

```
some_command
cfn-signal -e $? ...
```

You *must* always run `cfn-signal`, but then use it to send the error code of the previous command (typically but not necessarily this will be `cfn-init`). Of course, this breaks down and becomes complicated if you run multiple commands.

An alternative that I like better is to install a bash error trap:

```
set -euo pipefail
signal_failure() {
  cfn-signal --success false 
}
trap signal_failure ERR

command1
command2
command3
```

And this will do the right thing. But it is a lot more ceremony...

Reply:
Why not trap exit and just send the last status. If they choose not to exit on error, `strict` (I think). The signal behavior would be the same. This would simplify the need for two signal commands.

Reply:
Yes, I thought the same later, just hadn't figured out if I can use trap abd $! together in that way